crates/sargon/src/signing/petition_types/general_role_with_hd_factor_instance.rs

```diff
@@ -0,0 +1,229 @@
+use crate::prelude::*;
+
+decl_role_with_factors!(
+    /// A general depiction of each of the roles in a `MatrixOfFactorInstances`.
+    /// `SignaturesCollector` can work on any `RoleKind` when dealing with a securified entity.
+    General,
+    HierarchicalDeterministicFactorInstance
+);
+
+impl TryFrom<(MatrixOfFactorInstances, RoleKind)>
+    for GeneralRoleWithHierarchicalDeterministicFactorInstances
+{
+    type Error = CommonError;
+
+    fn try_from(value: (MatrixOfFactorInstances, RoleKind)) -> Result<Self> {
+        let matrix = value.0;
```

Comment:
you can write `let (matrix, role) = value;` I think, (or at very least you can write `(let matrix, let role) = value;`)